tests/unittests/Foundation/NSDictionaryTests.mm

```diff
@@ -130,3 +139,23 @@
 
     EXPECT_OBJCNE(input, output);
 }
+
+TEST(NSMutableDictionary, InsertingNilForKeyedSubscriptShouldRemoveValue) {
+    NSMutableDictionary* dict = [NSMutableDictionary dictionaryWithObject:@"world" forKey:@"hello"];
+    [dict setObject:nil forKeyedSubscript:@"hello"];
+    EXPECT_EQ(nil, dict[@"hello"]);
+    EXPECT_EQ(0, [dict count]);
+}
+
+TEST(NSMutableDictionary, InsertingNilWithLiteralSyntaxShouldRemoveValue) {
+    NSMutableDictionary* dict = [NSMutableDictionary dictionaryWithObject:@"world" forKey:@"hello"];
+    dict[@"hello"] = nil;
```

Comment:
"Literal syntax" only refers to `@{@"key": @"val"}`; you probably want Subscripting